tnnt/views.py

```diff
@@ -11,3 +26,310 @@ class AboutView(TemplateView):
 
 class ArchivesView(TemplateView):
     template_name = 'archives.html'
+
+class ClanMgmtView(View):
+    template_name = 'clanmgmt.html'
+
+    def get_context_data(self, **kwargs):
+        user = self.request.user
+        # look up their clan
+        kwargs['clan'] = None
+        try:
+            player = Player.objects.get(user=user.id)
+            kwargs['me'] = player
+            clan = player.clan
+            if clan is not None:
+                kwargs['clan'] = clan
+                kwargs['members'] = Player.objects.filter(clan=clan)
+                kwargs['invitees'] = clan.invitees.all()
+            kwargs['invites'] = player.invites.all()
+        except Player.DoesNotExist:
+            clan = None
+
+        if 'invite_member_form' not in kwargs:
+            kwargs['invite_member_form'] = InviteMemberForm()
+        if 'create_clan_form' not in kwargs:
+            kwargs['create_clan_form'] = CreateClanForm()
+
+        return kwargs
+
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return HttpResponseRedirect('/login')
+        return render(request, self.template_name, self.get_context_data(**kwargs))
+
+    def post(self, request, *args, **kwargs):
+
+        if not request.user.is_authenticated:
+            return HttpResponseRedirect('/login')
+
+        # if we passed the auth test above then this shouldn't fail...
+        player = Player.objects.get(user=request.user.id)
+        ctx = {}
+
+        if 'create_clan' in request.POST:
+            self.create_clan(request, player, ctx)
+
+        elif 'invite' in request.POST:
+            self.invite_to_clan(request, player, ctx)
+
+        elif 'leave' in request.POST:
+            self.leave_clan(request, player, ctx)
+
+        elif 'disband' in request.POST:
+            self.disband_clan(request, player, ctx)
+
+        elif 'join_clan' in request.POST:
+            self.join_clan(request, player, ctx)
+
+        elif 'rescind' in request.POST:
+            self.rescind_invite(request, player, ctx)
+
+        elif 'adminify' in request.POST:
+            self.make_admin(request, player, ctx)
+
+        elif 'kick' in request.POST:
+            self.kick_member(request, player, ctx)
+
+        return render(request, self.template_name, self.get_context_data(**ctx))
+
+    # Helper function triggered when a create_clan POST request comes in
+    def create_clan(self, request, player, ctx):
+        create_clan_form = CreateClanForm(request.POST)
+        new_clan_name = request.POST['clan_name']
```

Comment:
This will error if a user POSTs with 'create_clan' but no 'clan_name' in the request.

Reply:
shouldn't this use `create_clan_form.cleaned_data[]`?